test/presentation/user_action/user_action_details_view_model_test.dart

```diff
@@ -51,27 +56,83 @@ void main() {
       final store = Store<AppState>(
         reducer,
         initialState: AppState.initialState().copyWith(
+          userActionListState: UserActionListSuccessState([mockUserAction(id: 'id')]),
           userActionUpdateState: UserActionUpdatedState(UserActionStatus.DONE),
         ),
       );
 
       // When
-      final viewModel = UserActionDetailsViewModel.create(store);
+      final viewModel = UserActionDetailsViewModel.create(store, 'id');
 
       // Then
       expect(viewModel.displayState, UserActionDetailsDisplayState.SHOW_SUCCESS);
     });
   });
 
+  test("when action is on time", () {
+    // Given
+    final store = Store<AppState>(
+      reducer,
+      initialState: AppState.initialState().copyWith(
+        userActionListState: UserActionListSuccessState([mockUserAction(id: 'id', dateEcheance: DateTime(2041, 1, 1))]),
+        userActionUpdateState: UserActionUpdatedState(UserActionStatus.IN_PROGRESS),
+      ),
+    );
+
+    // When
+    final viewModel = UserActionDetailsViewModel.create(store, 'id');
+
+    // Then
+    expect(
+      viewModel.dateFormattedTexts,
+      [
+        FormattedText("À réaliser pour le "),
+        FormattedText("mardi 1 janvier", bold: true),
+      ],
+    );
+    expect(viewModel.dateBackgroundColor, AppColors.accent3Lighten);
+    expect(viewModel.dateTextColor, AppColors.accent2);
+    expect(viewModel.dateIcons, [Drawables.icClock]);
+  });
+
+  test("when action is late", () {
+    // Given
```

Comment:
Parel =>
"when action is late..... and what?" Qu'est-ce que fait le test?
Maybe "When action is late should show warning color and correct text"

Reply:
J'ai ajouté ta proposition ;)